### PR TITLE
Add/logging

### PR DESCRIPTION
**Pull request by leogermani**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Adds logging to contact syncing

### How to test the changes in this Pull Request:

See https://github.com/Automattic/newspack-manager/pull/240

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

**Comment by adekbadek** [comment]
@leogermani – just noting that [tests are not passing](https://app.circleci.com/pipelines/github/Automattic/newspack-newsletters/4687/workflows/3d881775-269d-4bec-823e-576768a0b509/jobs/18188/parallel-runs/0/steps/0-104). Should be fixed before a CR.

File: includes/service-providers/mailchimp/class-newspack-newsletters-mailchimp-notes.php [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+	 */
+	public static function handle_upsert( $provider, $contact, $lists, $result, $is_updating, $context ) {
+		if ( 'mailchimp' !== $provider ) {
+			return;
+		}
+
+		$lists_string = implode( ', ', $lists );
+		$message = sprintf(
+			/* translators: 1: email address, 2: list IDs */
+			__( 'Contact updated by Newspack. Context: %1$s. Lists added: %2$s', 'newspack-newsletters' ),
```
Comment (claudiulodro) [inline includes/service-providers/mailchimp/class-newspack-newsletters-mailchimp-notes.php]:
```suggestion
			__( 'Contact updated by Newspack from site %1$s. Context: %2$s. Lists added: %3$s', 'newspack-newsletters' ),
			get_site_url(),
```

**Review by claudiulodro — COMMENTED** [review]
Looks great! I think it's important to add the site URL to the MC notes, because there have been situations before where multiple sites were syncing to the same MC and causing weird data. This will make it easy to identify that situation.

File: includes/service-providers/mailchimp/class-newspack-newsletters-mailchimp-notes.php [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+		}
+
+		$lists_to_add_string = implode( ', ', $lists_to_add );
+		$lists_to_remove_string = implode( ', ', $lists_to_remove );
+
+		$message = sprintf(
+			/* translators: 1: email address, 2: lists added, 3: lists removed */
+			__( 'Contact updated by Newspack. Context: %1$s. Lists added: %2$s. Lists removed: %3$s', 'newspack-newsletters' ),
+			$context,
+			$lists_to_add_string,
```
Comment (claudiulodro) [inline includes/service-providers/mailchimp/class-newspack-newsletters-mailchimp-notes.php]:
```suggestion
			$lists_to_add_string ? $lists_to_add_string : '(none)',
```

Comment (claudiulodro) [reply includes/service-providers/mailchimp/class-newspack-newsletters-mailchimp-notes.php]: I'm not sure if `(none)` is the best when it's empty, but currently this comes through as e.g.
```
Contact updated by Newspack. Context: User updated their subscriptions on My Account page. Lists added: . Lists removed: 62688a1068
```
and
```
Contact updated by Newspack. Context: User updated their subscriptions on My Account page. Lists added: 62688a1068. Lists removed:
```

Comment (leogermani) [reply includes/service-providers/mailchimp/class-newspack-newsletters-mailchimp-notes.php]: addressed in https://github.com/Automattic/newspack-newsletters/pull/1574/commits/15ea318512af34a965810a64a9c916ebf453ecde

File: includes/class-newspack-newsletters-contacts.php [hunk 3]
```diff
@@ -139,7 +141,24 @@ public static function delete( $user_id ) {
...
+			'newspack_log',
+			'newspack_esp_sync_delete_contact',
+			$context,
+			[
+				'type'       => is_wp_error( $result ) ? 'error' : 'debug',
+				'data'       => [
+					'provider' => $provider->service,
+					'errors'   => is_wp_error( $result ) ? $result->get_error_message() : [],
+				],
+				'user_email' => $email,
```
Comment (dkoo) [inline includes/class-newspack-newsletters-contacts.php]:
```suggestion
				'user_email' => $user->user_email,
```

Getting a PHP warning: `PHP Warning:  Undefined variable $email in /srv/htdocs/wp-content/plugins/newspack-newsletters/includes/class-newspack-newsletters-contacts.php on line 156`

**Review by dkoo — CHANGES_REQUESTED** [review]
Only noticed one issue with an undefined variable upon deleting a user.

**Pull request merged**